sdk/lib/opentelemetry/sdk/resources/resource.rb

```diff
@@ -93,18 +99,36 @@ def attribute_enumerator
         end
 
         # Returns a new, merged {Resource} by merging the current {Resource} with
-        # the other {Resource}. In case of a collision, the current {Resource}
-        # takes precedence
+        # the other {Resource}. In case of an attributes collision, the other {Resource}
+        # takes precedence. If two incompatible schema urls are given, then the schema
+        # URL of the resulting resource will be unset.
         #
         # @param [Resource] other The other resource to merge
         # @return [Resource] A new resource formed by merging the current resource
         #   with other
-        def merge(other)
+        def merge(other) # rubocop:disable Metrics/AbcSize
           return self unless other.is_a?(Resource)
 
-          self.class.send(:new, attributes.merge(other.attributes).freeze)
+          # This is slightly verbose, but tries to follow the definition in the spec closely.
+          new_schema_url = if schema_url.nil?
+                             other.schema_url
+                           elsif other.schema_url.nil? || schema_url == other.schema_url
+                             schema_url
+                           elsif schema_url != other.schema_url
+                             # According to the spec: The resulting resource is undefined, and its contents are implementation-specific.
+                             # We choose to simply un-set the resource URL and log a warning about it, and allow the attributes to merge.
+                             OpenTelemetry.logger.warn(
+                               "Merging resources with schema version '#{schema_url}' and '#{other.schema_url}' is undefined."
+                             )
+
+                             nil
+                           end
+
+          self.class.send(:new, attributes.merge(other.attributes).freeze, new_schema_url.freeze)
         end
 
+        attr_reader :schema_url
```

Comment:
It is interesting to note that the spec doesn't say anything about a getter for `schema_url`. Obviously, we need one, since exporters need to be able to read it. We should open a PR against the spec to add this after https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#retrieve-attributes